qiskit/compiler/transpile.py

```diff
@@ -617,9 +618,17 @@ def _parse_translation_method(translation_method, num_circuits):
     return translation_method
 
 
-def _parse_scheduling_method(scheduling_method, num_circuits):
+def _parse_scheduling_method(scheduling_method, circuits):
+    """If there is a delay in any circuit, implicitly add a default scheduling method."""
+    if scheduling_method is None:
+        for circ in circuits:
+            for inst, _, _ in circ:
+                if isinstance(inst, Delay):
+                    scheduling_method = 'alap'
+                    break
+
```

Comment:
```suggestion
            else:
                continue
            break
```
maybe better to break from outer loop too.

Reply:
I just double checked; these are equivalent

Reply:
Lmk if you have a way to prove otherwise

Reply:
You're right that it doesn't break from the outer loop though. The alternative is:

```
def _parse_scheduling_method(scheduling_method, circuits):
    """If there is a delay in any circuit, implicitly add a default scheduling method."""
    delay_found = False
    if scheduling_method is None:
        for circ in circuits:
            if delay_found:
                break
            for inst, _, _ in circ:
                if isinstance(inst, Delay):
                    scheduling_method = 'alap'
                    delay_found = True
                    break
```

Reply:
Oh I just learned about this! Your suggestion works if it's unindented one level:
```
def _parse_scheduling_method(scheduling_method, circuits):
    """If there is a delay in any circuit, implicitly add a default scheduling method."""
    if scheduling_method is None:
        for circ in circuits:
            for inst, _, _ in circ:
                if isinstance(inst, Delay):
                    scheduling_method = 'alap'
                    break
            else:
                continue
            break
```

Reply:
I think it might be more readable to do:
```
def _parse_scheduling_method(scheduling_method, circuits):
    """If there is a delay in any circuit, implicitly add a default scheduling method."""
    def has_delay(circuit):
        for inst, _, _ in circuit:
            if isinstance(inst, Delay):
                return True
        return False

    if scheduling_method is None:
        for circ in circuits:
            if has_delay(circ):
                scheduling_method = 'alap'
                break
```

Reply:
Updated with the last one